### ISSUE TITLE
aws-sdk is required

### ISSUE THREAD

**Issue by hanxue** [kelektiv/node.bcrypt.js#607]
This is my code using `bcrypt`:

```js
import bcrypt from 'bcrypt'
//...
bcrypt.compare(this.form.password, dbPassword, function(err, res) {
          if (res) {
            // Rest of logging in code
          }
        })
```

I get an error during compile:

This dependency was not found:
```
* aws-sdk in ./node_modules/bcrypt/node_modules/node-pre-gyp/lib/info.js, ./node_modules/bcrypt/node_modul
es/node-pre-gyp/lib/publish.js and 1 other

To install it, you can run: npm install --save aws-sdk
```

Looking at `node-pre-gyp@0.9.1`'s `package.json`, aws-sdk is only specified as a devDependency

```js
 "dependencies": {
    "detect-libc": "^1.0.2",
    "mkdirp": "^0.5.1",
    "needle": "^2.2.0",
    "nopt": "^4.0.1",
    "npm-packlist": "^1.1.6",
    "npmlog": "^4.0.2",
    "rc": "^1.1.7",
    "rimraf": "^2.6.1",
    "semver": "^5.3.0",
    "tar": "^4"
  },
  "deprecated": false,
  "description": "Node.js native addon binary install tool",
  "devDependencies": {
    "aws-sdk": "^2.28.0",
    "jshint": "^2.9.5",
    "nock": "^9.2.3",
    "retire": "^1.2.12",
    "tape": "^4.6.3"
  },
```

Not quite sure why node needs aws-sdk when I am not building anything with node-pre-gyp. Is there anyway I can avoid installing aws-sdk?

### Environment

OS: MacOS 10.10.5
Node: 9.11.1
Vue.js: 2.5.2
Yarn: 1.6.0

**Comment by agathver** [comment]
It appears you are trying to use bcrypt in a JS bundler such as Webpack. `bcrypt` is not designed to be run there. `bcrypt` is a native (C++) module and only runs on `node` environments.

You can use `bcrypt.js` as a (mostly) drop-in replacement.

**Comment by grzegorzjudas** [comment]
I just had this exact same error, for whatever reason. @agathver, not exactly sure what you mean bcrypt is not designed to be run by webpack - it's a node library, isn't it? And most of projects nowadays use webpack... So it's fair to assume bcrypt should bundle without errors as 99% of other libraries do.

**Comment by agathver** [comment]
Not all NodeJS modules can be used in frontend. You can't run C++ code in a browser. It is a node module but it isn't a JS module.

**Comment by grzegorzjudas** [comment]
Right, but I do run it in node only. Actually, this is an express backend.
